website/source/docs/job-specification/spread.html.md

```diff
@@ -0,0 +1,117 @@
+---
+layout: "docs"
+page_title: "spread Stanza - Job Specification"
+sidebar_current: "docs-job-specification-spread"
+description: |-
+  The "spread" stanza allows operators to configure desired target percentages for allocations according to node attributes or metadata.
+  Spread may be specified at the job, group, or task levels for ultimate flexibility.
+  More than one spread stanza may be specified with relative weights between each.
+---
+
+# `spread` Stanza
+
+<table class="table table-bordered table-striped">
+  <tr>
+    <th width="120">Placement</th>
+    <td>
+      <code>job -> **spread**</code>
+      <br>
+      <code>job -> group -> **spread**</code>
+      <br>
+      <code>job -> group -> task -> **spread**</code>
+    </td>
+  </tr>
+</table>
+
+The `spread` stanza allows operators to specify target percentages of allocations according to
+specific node attributes or metadata. Spread may
+be expressed on [attributes][interpolation] or [client metadata][client-meta].
+Additionally spread may be specified at the [job][job], [group][group], or
+[task][task] levels for ultimate flexibility.
+
+Nomad will use any defined spread criteria when computing node scores for placement.
+Nodes are scored acccording to how closely they match the desired target percentage defined in the
+spread stanza. Spread scores are combined with other scoring factors such as bin packing.
+
+A job or task group can have more than one spread criteria, with weights to express relative preference.
+
+Spread criteria are treated as a soft preference by the Nomad scheduler. If no nodes match a given spread criteria, placement is still successful.
+
+## `spread` Parameters
+
+- `attribute` `(string: "")` - Specifies the name or reference of the attribute
+  to use. This can be any of the [Nomad interpolated
+  values](/docs/runtime/interpolation.html#interpreted_node_vars).
+
+- `target` <code>([target](#target-parameters): <required>)</code> - Specifies one or more target
+   percentages for each value of the `attribute` in the spread stanza. If this is omitted,
+   Nomad will spread allocations evenly across all values of the attribute.
+
+- `weight` `(integer:0)` - Specifies a weight for the spread stanza. The weight is used
```

Comment:
I thought the weight of spread is only 0-100

Reply:
https://github.com/hashicorp/nomad/blob/master/nomad/structs/structs.go#L6567

Reply:
I also went to look what the default is for spread's weight and it is zero and the api is such that the int is not a pointer which means we can't detect unset vs zero: https://github.com/hashicorp/nomad/blob/6b7953af21417c28c8f4d78e7a031f86be0d842d/api/tasks.go#L226.

I think it would be good to change this to a pointer and make the default value 50